File: backend/app/schemas/world.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from datetime import datetime

class WorldMeta(BaseModel):
    world_id: str
    title: Optional[str] = Field(None, description="自定义世界标题")
    name: str = Field(..., description="核心场景名称 (fallback)")
    era: str = Field(..., description="时代背景")
    created_at: datetime = Field(default_factory=datetime.now)
    npc_count: int = Field(default=0)
    preview: str = Field(default="", description="简短描述")

class WorldListResponse(BaseModel):
    worlds: List[WorldMeta]

class WorldConfig(BaseModel):
    is_illustrated: bool = False
    enable_advanced_tasks: bool = False
    image_style: Optional[str] = None
    manga_page_size: int = 10
    world_confirmed: bool = False
    roster_confirmed: bool = False
    quest_confirmed: bool = False

class LoadWorldResponse(BaseModel):
    status: str
    world_bible: Dict[str, Any]
    npcs: List[Dict[str, Any]]
    items: List[Dict[str, Any]] = []
    locations: List[Dict[str, Any]] = []
    time_config: Optional[Dict[str, Any]] = None
    quests: List[Dict[str, Any]] = []
    chat_history: List[Dict[str, Any]]
    schedules: Dict[str, List[Dict[str, Any]]] = {}
    is_locked: bool = False
    config: WorldConfig = Field(default_factory=WorldConfig)

class WorldConfigUpdate(BaseModel):
    config: WorldConfig

File: frontend/src/api/world.ts
import axios from 'axios';
import { WorldBible, NPC, Quest } from './genesis';

const api = axios.create({
    baseURL: '/api/worlds',
    headers: {
        'Content-Type': 'application/json',
    },
});

export interface WorldMeta {
    world_id: string;
    title?: string;
    name: string;
    era: string;
    created_at: string;
    npc_count: number;
    preview: string;
}

export interface WorldConfig {
    is_illustrated: boolean;
    enable_advanced_tasks: boolean;
    image_style?: string;
    manga_page_size?: number;
    world_confirmed?: boolean;
    roster_confirmed?: boolean;
    quest_confirmed?: boolean;
}

export interface LoadWorldResponse {
    status: string;
    world_bible: WorldBible;
    npcs: NPC[];
    items?: any[];
    locations?: any[];
    time_config?: any;
    quests?: Quest[];
    chat_history: any[];
    schedules?: Record<string, any[]>;
    is_locked?: boolean;
    config: WorldConfig;
}

export default {
    async listWorlds() {
        const response = await api.get<{ worlds: WorldMeta[] }>('/');
        return response.data;
    },

    async loadWorld(worldId: string) {
        // Add cache busting timestamp
        const response = await api.get<LoadWorldResponse>(`/${worldId}?t=${Date.now()}`);
        return response.data;
    },

    async updateConfig(worldId: string, config: WorldConfig) {
        const response = await api.patch(`/${worldId}/config`, { config });
        return response.data;
    },

    async updateTitle(worldId: string, title: string) {
        const response = await api.patch<{ status: string; title: string }>(`/${worldId}/title`, { title });
        return response.data;
    },

    async deleteWorld(worldId: string) {
        const response = await api.delete(`/${worldId}`);
        return response.data;
    },
};
